test/Microsoft.AspNetCore.SignalR.Test.Server/wwwroot/js/hubConnectionTests.js

```diff
@@ -0,0 +1,79 @@
+const TESTHUBENDPOINT_URL = `http://${document.location.host}/testhub`;
+
+describe('hubConnection', () => {
+    eachTransport(transportName => {
+        it(`over ${transportName} can invoke server method and receive result`, done => {
+            const message = "Hi";
+            let hubConnection = new signalR.HubConnection(TESTHUBENDPOINT_URL, 'formatType=json&format=text');
+
+            hubConnection.start(transportName)
+                .then(() => {
+                    hubConnection.invoke('Microsoft.AspNetCore.SignalR.Test.Server.TestHub.Echo', message)
+                    .then(result => {
+                        expect(result).toBe(message);
+                    })
+                    .catch(() => {
+                        expect(false).toBe(true);
+                    })
+                    .then(() => {
+                        hubConnection.stop();
+                        done();
+                    })
+                })
+                .catch(() => {
+                    expect(false).toBe(true);
+                    done();
+                })
+        });
+
+        it(`over ${transportName} rethrows an exception from the server`, done => {
+            const errorMessage = "An error occurred.";
+            let hubConnection = new signalR.HubConnection(TESTHUBENDPOINT_URL, 'formatType=json&format=text');
+
+            hubConnection.start(transportName)
+                .then(() => {
+                    hubConnection.invoke('Microsoft.AspNetCore.SignalR.Test.Server.TestHub.ThrowException', errorMessage)
+                        .then(() => {
+                            // exception expected but none thrown
+                            expect(true).toBe(false);
+                        })
+                        .catch(e => {
+                            expect(e.message).toBe(errorMessage);
```

Comment:
Not really. Exception types from one language don't translate well to exception types in another language and trying to force it would be weird. So we just throw `Error` in javascript. We also don't even send any exception details at the moment (stacktrace and maybe even the exception type might be useful for debugging/tracing)